src/tasks/WasmAppBuilder/WasmAppBuilder.cs

```diff
@@ -168,26 +71,40 @@ protected override bool ExecuteInternal()
         }
         MainAssemblyName = Path.GetFileName(MainAssemblyName);
 
-        var config = new WasmAppConfig()
+        var config = new BootJsonData()
         {
-            MainAssemblyName = MainAssemblyName,
-            GlobalizationMode = InvariantGlobalization ? "invariant" : HybridGlobalization ? "hybrid" : "icu"
+            config = new(),
+            entryAssembly = MainAssemblyName,
+            icuDataMode = InvariantGlobalization
+                ? ICUDataMode.Invariant
+                : !string.IsNullOrEmpty(WasmIcuDataFileName)
+                    ? ICUDataMode.Custom
+                    : HybridGlobalization
+                        ? ICUDataMode.Hybrid
+                        : WasmIncludeFullIcuData
+                            ? ICUDataMode.All
+                            : ICUDataMode.Sharded
```

Comment:
Is this covered by tests?

Reply:
Yes, learned the hard way, I was failing before I got it all right